pkg/kubelet/kubelet.go

```diff
@@ -1664,14 +1667,44 @@ func getHostPortConflicts(pods []api.Pod) []api.Pod {
 	return conflicts
 }
 
-// handleHostPortConflicts handles pods that conflict on Port.HostPort values.
-func (kl *Kubelet) handleHostPortConflicts(pods []api.Pod) {
+func (kl *Kubelet) getPodsExceedingCapacity(pods []api.Pod) []api.Pod {
+	info, err := kl.GetCachedMachineInfo()
+	if err != nil {
+		glog.Error("error getting machine info: %v", err)
+		return []api.Pod{}
+	}
+
+	// Respect the pod creation order when resolving conflicts.
+	sort.Sort(podsByCreationTime(pods))
+
+	capacity := CapacityFromMachineInfo(info)
+	return scheduler.GetPodsExceedingCapacity(pods, capacity)
+}
+
+// handleNotfittingPods handles pods that do not fit on the node.
+// Currently conflicts on Port.HostPort values and exceeding node capacity are handled.
+func (kl *Kubelet) handleNotfittingPods(pods []api.Pod) {
 	conflicts := getHostPortConflicts(pods)
+	conflictsMap := map[types.UID]bool{}
 	for _, pod := range conflicts {
 		kl.recorder.Eventf(&pod, "hostPortConflict", "Cannot start the pod due to host port conflict.")
 		kl.setPodStatusInCache(GetPodFullName(&pod), api.PodStatus{
 			Phase:   api.PodFailed,
 			Message: "Pod cannot be started due to host port conflict"})
+		conflictsMap[pod.UID] = true
+	}
+	remainingPods := []api.Pod{}
+	for _, pod := range pods {
+		if !conflictsMap[pod.UID] {
+			remainingPods = append(remainingPods, pod)
+		}
+	}
+	conflicts = kl.getPodsExceedingCapacity(remainingPods)
+	for _, pod := range conflicts {
+		kl.recorder.Eventf(&pod, "capacityExceeded", "Cannot start the pod due to exceeded capacity.")
+		kl.setPodStatusInCache(GetPodFullName(&pod), api.PodStatus{
+			Phase:   api.PodFailed,
+			Message: "Pod cannot be started due to exceeded capacity"})
```

Comment:
Ideally, would be useful to indicate which resource, and by how much (think "why pending").